build/build.go

```diff
@@ -81,14 +85,21 @@ func importWithSrcDir(path string, srcDir string, mode build.ImportMode, install
 			buildContext.InstallSuffix += "_" + installSuffix
 		}
 	}
+
+	// if current path is for gopherjs import and a vendored path for gopherjs has previously been encountered, update
+	// the path to be the vendored one
+	if strings.HasPrefix(path, "github.com/gopherjs/gopherjs") && vendoredGopherJS != "" {
+		path = vendoredGopherJS + strings.TrimPrefix(path, "github.com/gopherjs/gopherjs")
+	}
+
 	pkg, err := buildContext.Import(path, srcDir, mode)
 	if err != nil {
 		return nil, err
 	}
 
-	// TODO: Resolve issue #415 and remove this temporary workaround.
+	// if gopherjs package path is vendored, store it so that we can update future paths
 	if strings.HasSuffix(pkg.ImportPath, "/vendor/github.com/gopherjs/gopherjs/js") {
-		return nil, fmt.Errorf("vendoring github.com/gopherjs/gopherjs/js package is not supported, see https://github.com/gopherjs/gopherjs/issues/415")
+		vendoredGopherJS = strings.TrimSuffix(pkg.ImportPath, "/js")
```

Comment:
Agree that, even though this appears to be true by convention right now, it is brittle and we shouldn't depend on it.

Conceptually, when a `build` is executed, we want all the `gopherjs` imports to resolve to some import path (either the canonical one or the vendored one). I think the most common case is that the `github.com/gopherjs/gopherjs/js` import path that is resolved from the location of the input file(s) is the correct one. Given this, I think I would advocate for something like the following:

* When `build` is first invoked, resolve the `github.com/gopherjs/gopherjs/js` import from the location of the first argument (current working directory if none is specified)
  * There is an edge case here of what to do if multiple input files in different directories are specified -- I think it's probably reasonable to default to import path relative to the first argument for now. If it's really necessary, we could also introduce a flag that allows the path to the vendored `gopherjs` to be explicitly specified
* Pass that in as a parameter to this and other functions that need to do import resolution

This makes the entire approach more stateless and explicit based on parameters (and if the vendored path is empty, then we simply don't do any re-writes as the default behavior)